app/controllers/documents_controller.rb

```diff
@@ -0,0 +1,44 @@
+class DocumentsController < ApplicationController
+  def create
+    document = DocumentUpload.new(document_param)
+
+    error_msg = if document.valid?
+                  response = document.upload!(collection_ref: current_files_collection_ref)
+                  'There was an error uploading the file. Please try again.' if response.error?
+                else
+                  # TODO: how to deal with these errors?
+                  "There were errors: #{document.errors}"
+                end
+
+    redirect_back(fallback_location: root_path, alert: error_msg)
```

Comment:
Should we perhaps add a hidden field to the form(s) to hold the controller name we just came from? I'm usually very cautious about `redirect_back` because of firewalls, browser plugins, etc. blocking the referrer and it seems a bit bad UX to kick the user back to the homepage!